### ISSUE TITLE
[UX] Support for dropping catalog items on the canvas

### ISSUE THREAD

**Issue by lhein** [KaotoIO/kaoto-ui#766]
**Is your feature request related to a problem? Please describe.**
When starting with a blank project you need to drop catalog items directly on existing figures on the canvas. This is very non intuitive for users because you would normally expect to just drop something anywhere on the canvas and it will be added automatically either as the first step in a blank integration or as the last step (if valid) if there are already other steps available. 

**Describe the solution you'd like**
Dropping a source on a blank canvas should be allowed and the step should be made the first step.
Dropping another source if there is already a source should cause an error and shouldn't be allowed.
Adding an action or sink on a blank canvas shouldn't be allowed and spawn an error.
Adding an action or sink if there is a source or another action already should add the new item as the next step in the chain.

**Comment by kahboom** [comment]
It's slightly more complicated than it sounds to implement this.

We previously supported this as "temporary steps" that are not synchronized back to the YAML editor and never "make it" to the API, so that users could configure the steps individually while planning their integrations, before connecting them.

The problem with this is that we would need to persist steps individually as their own integration, because we don't have a persistence layer. So, each dropped step would need to be an integration, and how would we "know" which one to synchronize the YAML with?

This is why I previously had a designated area that would sync with the YAML, and anything outside of that would not. I'm not sure if you remember, but I had shared this as a possible example: https://flow-dnd.netlify.app/

Furthermore, #609 should be completed beforehand, but doesn't have to be if we don't allow the user to configure steps individually. Some planning and additional considerations would need to go into this feature.

**Comment by lhein** [comment]
I am not sure if we talk about the same thing. Having unconnected nodes on the canvas is not what I wanted. For instance if you have a clean canvas and drop a source it will be the first step in the integration. When now dropping an action or a sink on the canvas it will be automatically connected to the last figure on canvas (which is the source in this case).

**Comment by kahboom** [comment]
Okay so that would only be for the very initial state and only the first/second node, or if they are dropping them in a valid order (start + middle + middle + end), correct? As any other time wouldn't make sense I think (for example, if you already have connected nodes).

**Comment by kahboom** [comment]
Hmm in that case it would make sense to rename this issue to ".. when visualization/canvas is empty", and create a separate issue for inserting steps via drag and drop, as I'm trying to keep issues smaller and more specific.

**Comment by lhein** [comment]
No, this would change the context of my issue quite too much. It is not just about dropping the first step, it is a general request to allow this. If you would like to avoid the drop on canvas topic we should consider removing the catalog from the top menu and start with something like a big plus icon and use the contextual catalog from there.

**Comment by kahboom** [comment]
Just following up on this topic--dropping a step onto an empty canvas or as a first step (even if it's to replace an existing step) is actually a different implementation than inserting steps in between existing ones by dropping them onto the connection. The second one is an enhancement on inserting steps, whereas the first is an enhancement on replacing steps (even the initial step is considered a step replacement).

**Comment by lhein** [comment]
@kahboom I think this one can be closed. I like the interaction with the mini catalog more. We should however maybe think about using the left side catalog for templates at some point.
